### PR TITLE
MAINT: stronger test for returned Jmin from `differential_evolution`

### PR DESCRIPTION
**Pull request by andyfaff**

### CODE REVIEW

File: scipy/optimize/tests/test__differential_evolution.py [hunk 1]
```diff
@@ -231,9 +231,11 @@ def test__ensure_constraint(self):
     def test_differential_evolution(self):
         # test that the Jmin of DifferentialEvolutionSolver
         # is the same as the function evaluation
-        solver = DifferentialEvolutionSolver(self.quadratic, [(-2, 2)])
+        solver = DifferentialEvolutionSolver(
+            self.quadratic, [(-2, 2)], maxiter=1, polish=False
+        )
         result = solver.solve()
-        assert_almost_equal(result.fun, self.quadratic(result.x))
+        assert_equal(result.fun, self.quadratic(result.x))
```
Comment (andyfaff) [inline scipy/optimize/tests/test__differential_evolution.py:238]: They should be exactly the same value

**Comment by mdhaber** [comment]
Seems fine if tests pass. What's the motivation for turning `polish` off now? Is setting the iteration limit just to reduce the time it takes?

Lint failure looks unrelated.

**Comment by andyfaff** [comment]
THe test is checking that `np.testing.assert_equal(fun(OptimizeResult.x), OptimizeResult.fun)`. The function is a quadratic, so if polishing is done the final function value will be very close to 0, and I wanted to have a value further away.

You've now made me realise that it should probably check with polishing on or off.

**Comment by andyfaff** [comment]
Just tightened the test further. It's to prove #15594 is not an issue.

**Comment by mdhaber** [comment]
OK, when tests finish I'll merge.

**Pull request merged**